text-generation-inference/server/text_generation_server/generator.py

```diff
@@ -646,3 +599,170 @@ def from_pretrained(
+    # TODO: maybe model_config can be removed from mailbox
+
+    def return_to_caller(*data):
+        # consider adding a rendezvous here
+        if rank == 0:
+            xm.mark_step()
+            mailbox.send(*data)
+
+
+    while True:
+        xm.rendezvous("start")
+        if rank == 0:
+            mailbox.agent_ready.set()
+            mailbox.receive()
+        # Wait for rank 0 to receive command
+        xm.rendezvous("wait_command")
+        command, data = mailbox.command_data
+        logger.debug(f"Generator@{rank} {command.name}")
+        if command == GeneratorCommand.INFO:
+            info = generator.info
+            return_to_caller(info.SerializeToString())
+        if command == GeneratorCommand.WARMUP:
+            batch = Batch.FromString(data[0])
+            return_to_caller(generator.warmup(batch=batch))
+        if command == GeneratorCommand.PREFILL:
+            batch = Batch.FromString(data[0])
+            generations, cached_batch = generator.prefill(batch=batch)
+            return_to_caller([g.SerializeToString() for g in generations], cached_batch.SerializeToString())
+        if command == GeneratorCommand.DECODE:
+            batches = [CachedBatch.FromString(b) for b in data[0]]
+            generations, cached_batch = generator.decode(batches=batches)
+            s_cached_batch = cached_batch.SerializeToString() if cached_batch is not None else None
+            return_to_caller([g.SerializeToString() for g in generations], s_cached_batch)
+        if command == GeneratorCommand.FILTER:
+            batch_id, request_ids = data
+            cached_batch = generator.filter(batch_id, request_ids)
+            return_to_caller(cached_batch.SerializeToString())
+        if command == GeneratorCommand.CLEAR:
+            generator.clear()
+        if command == GeneratorCommand.DELETE:
+            if rank == 0:
+                # Set agent to ready
+                mailbox.agent_ready.set()
+            break
+
+
+def model_loop_fn(*args):
+    """Spawn processes in the TPUs forwarding arguments"""
+    xmp.spawn(_mp_fn, args=(args), join=True, daemon=False)
+
+class TpuGenerator(Generator):
+    """A Generator for models running on TPU.
+
+    This generator actually spawns several processes to handle the requests in sharded models whenever possible.
+    """
+
+    def __init__(self, model_path: str, revision: str, max_batch_size: int, max_sequence_length: int):
+        manager = mp.Manager()
+        self.mailbox = RootMailbox(manager)
+
+        # Disable parallelism on tokenizers to avoid deadlocks on TPU threads
+        os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+        self.model_loop = mp.Process(
+            target=model_loop_fn, args=(model_path, revision, max_batch_size, max_sequence_length, self.mailbox)
+        )
+        self.model_loop.start()
+
+    @property
+    def info(self) -> InfoResponse:
+        s_info = self.mailbox.send(GeneratorCommand.INFO, None)[0]
```

Comment:
```suggestion
        s_info = self.mailbox.send(GeneratorCommand.INFO, None)[0]
        s_info, _ = self.mailbox.send(GeneratorCommand.INFO, None)
```

Reply:
This cannot be done, because `info` will only return one element, so the command returns a list with one value in it.